{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "WARNING: The MAMBA_ROOT_PREFIX environment variable is not set.\n",
            "WARNING: This is required for mamba to work correctly as of 2.0.\n",
            "WARNING: \n",
            "WARNING: For now, we are setting 'MAMBA_ROOT_PREFIX' to '/usr'.\n",
            "WARNING: \n",
            "WARNING: Please make sure this is consistent with your installation or alternatively (by order of preference):\n",
            "WARNING:   - rerun 'mamba shell init' to initialize mamba for your current shell\n",
            "WARNING:   - manually set 'MAMBA_ROOT_PREFIX' to the root of your installation in your shell profile script.\n",
            "WARNING:   - use the '-r,--root-prefix' CLI option when calling mamba.\n",
            "WARNING: \n",
            "WARNING: This message originates from /usr/etc/profile.d/mamba.sh\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "cd ..\n",
        "source ./activateEnv.sh\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Pinned packages:\n",
            "\n",
            "  - python=3.10\n",
            "\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "error    libmamba Could not create directory '/usr/pkgs': Permission denied\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Transaction\n",
            "\n",
            "  Prefix: /home/figa/Documents/Programming/VSCode/Metagenomics-Course/env\n",
            "\n",
            "  All requested packages already installed\n",
            "\n",
            "\n",
            "Transaction starting\n",
            "\n",
            "Transaction finished\n",
            "\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Instalar os software que vamos usar nesta prÃ¡tica\n",
        "mamba install -c bioconda -c conda-forge fastqc bwa gatk4 cutadapt samtools bedtools ensembl-vep=115 -y"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "%%bash\n",
        "\n",
        "mkdir -p fastq bwa bed cutadapt raw reference gatk vep \n",
        "mv *.fastq.gz raw/\n",
        "gunzip -c Chr7-reference.fasta.gz > reference/Chr7-reference.fasta\n",
        "rm Chr7-reference.fasta.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "application/gzip\n",
            "application/gzip\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Started analysis of caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 5% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 10% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 15% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 20% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 25% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 30% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 35% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 40% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 45% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 50% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 55% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 60% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 65% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 70% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 75% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 80% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 85% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 90% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 95% complete for caso3-tumor-pulmonar_R1.fastq.gz\n",
            "Approx 100% complete for caso3-tumor-pulmonar_R1.fastq.gz\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Analysis complete for caso3-tumor-pulmonar_R1.fastq.gz\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Started analysis of caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 5% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 10% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 15% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 20% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 25% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 30% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 35% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 40% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 45% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 50% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 55% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 60% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 65% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 70% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 75% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 80% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 85% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 90% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 95% complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Approx 100% complete for caso3-tumor-pulmonar_R2.fastq.gz\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Analysis complete for caso3-tumor-pulmonar_R2.fastq.gz\n",
            "This is cutadapt 5.1 with Python 3.10.14\n",
            "Command line parameters: -u 5 -U 5 -u -5 -U -5 -m 90 -o cutadapt/caso3-tumor-pulmonar-trimmed_R1.fastq.gz -p cutadapt/caso3-tumor-pulmonar-trimmed_R2.fastq.gz raw/caso3-tumor-pulmonar_R1.fastq.gz raw/caso3-tumor-pulmonar_R2.fastq.gz\n",
            "Processing paired-end reads on 1 core ...\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total read pairs processed:          4,000,000\n",
            "\n",
            "== Read fate breakdown ==\n",
            "Pairs that were too short:                   1 (0.0%)\n",
            "Pairs written (passing filters):     3,999,999 (100.0%)\n",
            "\n",
            "Total basepairs processed: 1,199,968,732 bp\n",
            "  Read 1:   599,968,852 bp\n",
            "  Read 2:   599,999,880 bp\n",
            "Total written (filtered):  1,119,968,504 bp (93.3%)\n",
            "  Read 1:   559,968,763 bp\n",
            "  Read 2:   559,999,741 bp\n"
          ]
        }
      ],
      "source": [
        "%%bash \n",
        "fastqc -o fastq/ raw/caso3-tumor-pulmonar_R1.fastq.gz raw/caso3-tumor-pulmonar_R2.fastq.gz\n",
        "cutadapt \\\n",
        "  -u 5 -U 5 \\\n",
        "  -u -5 -U -5 \\\n",
        "  -m 90 \\\n",
        "  -o cutadapt/caso3-tumor-pulmonar-trimmed_R1.fastq.gz \\\n",
        "  -p cutadapt/caso3-tumor-pulmonar-trimmed_R2.fastq.gz \\\n",
        "  raw/caso3-tumor-pulmonar_R1.fastq.gz \\\n",
        "  raw/caso3-tumor-pulmonar_R2.fastq.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "[bwa_index] Pack FASTA... 0.31 sec\n",
            "[bwa_index] Construct BWT for the packed sequence...\n",
            "[BWTIncCreate] textLength=318691946, availableWord=34423912\n",
            "[BWTIncConstructFromPacked] 10 iterations done. 56784042 characters processed.\n",
            "[BWTIncConstructFromPacked] 20 iterations done. 104904522 characters processed.\n",
            "[BWTIncConstructFromPacked] 30 iterations done. 147670090 characters processed.\n",
            "[BWTIncConstructFromPacked] 40 iterations done. 185676202 characters processed.\n",
            "[BWTIncConstructFromPacked] 50 iterations done. 219452122 characters processed.\n",
            "[BWTIncConstructFromPacked] 60 iterations done. 249468234 characters processed.\n",
            "[BWTIncConstructFromPacked] 70 iterations done. 276142554 characters processed.\n",
            "[BWTIncConstructFromPacked] 80 iterations done. 299846746 characters processed.\n",
            "[bwt_gen] Finished constructing BWT in 89 iterations.\n",
            "[bwa_index] 45.71 seconds elapse.\n",
            "[bwa_index] Update BWT... 0.32 sec\n",
            "[bwa_index] Pack forward-only FASTA... 0.17 sec\n",
            "[bwa_index] Construct SA from BWT and Occ... 26.96 sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa index reference/Chr7-reference.fasta\n",
            "[main] Real time: 74.405 sec; CPU: 73.464 sec\n",
            "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
            "[M::process] read 71448 sequences (10000073 bp)...\n",
            "[M::process] read 71410 sequences (10000248 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32010, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.70, 28.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71448 reads in 6.026 CPU sec, 5.962 real sec\n",
            "[M::process] read 71410 sequences (10000084 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32091, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.01, 28.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 6.088 CPU sec, 6.020 real sec\n",
            "[M::process] read 71414 sequences (10000190 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31985, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.97, 28.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 5.995 CPU sec, 5.928 real sec\n",
            "[M::process] read 71400 sequences (10000121 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32109, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.56, 28.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71414 reads in 5.997 CPU sec, 5.930 real sec\n",
            "[M::process] read 71448 sequences (10000058 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32154, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.82, 28.17)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71400 reads in 5.931 CPU sec, 5.863 real sec\n",
            "[M::process] read 71464 sequences (10000063 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32158, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.21, 28.43)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71448 reads in 5.953 CPU sec, 5.890 real sec\n",
            "[M::process] read 71466 sequences (10000037 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32021, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.93, 28.11)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71464 reads in 5.877 CPU sec, 5.812 real sec\n",
            "[M::process] read 71410 sequences (10000141 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32121, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.68, 28.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71466 reads in 5.936 CPU sec, 5.870 real sec\n",
            "[M::process] read 71430 sequences (10000086 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32062, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.00, 28.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 5.952 CPU sec, 5.885 real sec\n",
            "[M::process] read 71428 sequences (10000244 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31997, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 5.929 CPU sec, 5.862 real sec\n",
            "[M::process] read 71398 sequences (10000275 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32124, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.96, 28.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 5.946 CPU sec, 5.877 real sec\n",
            "[M::process] read 71444 sequences (10000074 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32087, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.01, 28.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71398 reads in 5.925 CPU sec, 5.861 real sec\n",
            "[M::process] read 71424 sequences (10000040 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32103, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.85, 28.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71444 reads in 5.969 CPU sec, 5.902 real sec\n",
            "[M::process] read 71438 sequences (10000203 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32067, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71424 reads in 5.781 CPU sec, 5.705 real sec\n",
            "[M::process] read 71446 sequences (10000275 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32054, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.86, 28.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 5.678 CPU sec, 5.602 real sec\n",
            "[M::process] read 71450 sequences (10000246 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32150, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.77, 28.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71446 reads in 5.704 CPU sec, 5.626 real sec\n",
            "[M::process] read 71406 sequences (10000181 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32111, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.94, 28.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71450 reads in 5.719 CPU sec, 5.643 real sec\n",
            "[M::process] read 71426 sequences (10000133 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32125, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.79, 28.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71406 reads in 5.636 CPU sec, 5.565 real sec\n",
            "[M::process] read 71454 sequences (10000196 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32129, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.17, 28.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 5.704 CPU sec, 5.627 real sec\n",
            "[M::process] read 71408 sequences (10000131 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32125, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.63, 28.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71454 reads in 5.663 CPU sec, 5.590 real sec\n",
            "[M::process] read 71410 sequences (10000107 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31999, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.00, 28.52)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71408 reads in 5.657 CPU sec, 5.580 real sec\n",
            "[M::process] read 71456 sequences (10000124 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32120, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
            "[M::mem_pestat] mean and std.dev: (340.04, 28.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 5.725 CPU sec, 5.650 real sec\n",
            "[M::process] read 71440 sequences (10000064 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32140, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 338, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.64, 28.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71456 reads in 5.662 CPU sec, 5.586 real sec\n",
            "[M::process] read 71420 sequences (10000207 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32121, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.02, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71440 reads in 5.720 CPU sec, 5.643 real sec\n",
            "[M::process] read 71406 sequences (10000073 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32151, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.97, 28.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 5.683 CPU sec, 5.607 real sec\n",
            "[M::process] read 71432 sequences (10000093 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32037, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.83, 28.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71406 reads in 5.665 CPU sec, 5.591 real sec\n",
            "[M::process] read 71470 sequences (10000115 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32079, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.99, 28.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71432 reads in 5.673 CPU sec, 5.597 real sec\n",
            "[M::process] read 71432 sequences (10000099 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32334, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.85, 28.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71470 reads in 5.662 CPU sec, 5.584 real sec\n",
            "[M::process] read 71462 sequences (10000206 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32060, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.98, 28.16)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71432 reads in 5.650 CPU sec, 5.573 real sec\n",
            "[M::process] read 71410 sequences (10000184 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31966, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.90, 28.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71462 reads in 5.681 CPU sec, 5.605 real sec\n",
            "[M::process] read 71440 sequences (10000197 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32192, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.13, 28.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 5.664 CPU sec, 5.587 real sec\n",
            "[M::process] read 71450 sequences (10000096 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31959, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.89, 28.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71440 reads in 5.722 CPU sec, 5.644 real sec\n",
            "[M::process] read 71418 sequences (10000048 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32098, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.66, 28.45)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71450 reads in 5.751 CPU sec, 5.678 real sec\n",
            "[M::process] read 71464 sequences (10000060 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32101, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.91, 28.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71418 reads in 5.655 CPU sec, 5.581 real sec\n",
            "[M::process] read 71418 sequences (10000096 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32013, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.23, 28.47)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71464 reads in 5.674 CPU sec, 5.598 real sec\n",
            "[M::process] read 71452 sequences (10000142 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32140, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.88, 28.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71418 reads in 5.715 CPU sec, 5.646 real sec\n",
            "[M::process] read 71430 sequences (10000027 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32179, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71452 reads in 5.668 CPU sec, 5.590 real sec\n",
            "[M::process] read 71422 sequences (10000050 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32126, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.05, 28.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 5.646 CPU sec, 5.566 real sec\n",
            "[M::process] read 71434 sequences (10000184 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32051, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.92, 28.41)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71422 reads in 5.703 CPU sec, 5.630 real sec\n",
            "[M::process] read 71418 sequences (10000301 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32170, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.84, 28.13)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71434 reads in 5.671 CPU sec, 5.596 real sec\n",
            "[M::process] read 71468 sequences (10000209 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32050, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.94, 28.46)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71418 reads in 5.662 CPU sec, 5.589 real sec\n",
            "[M::process] read 71410 sequences (10000167 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32061, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.82, 28.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71468 reads in 5.694 CPU sec, 5.617 real sec\n",
            "[M::process] read 71406 sequences (10000047 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32058, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.63, 28.17)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 5.651 CPU sec, 5.581 real sec\n",
            "[M::process] read 71464 sequences (10000048 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32161, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.84, 28.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71406 reads in 5.843 CPU sec, 5.775 real sec\n",
            "[M::process] read 71438 sequences (10000063 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32193, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.21, 28.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71464 reads in 5.952 CPU sec, 5.884 real sec\n",
            "[M::process] read 71432 sequences (10000117 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32180, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.83, 28.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 5.859 CPU sec, 5.790 real sec\n",
            "[M::process] read 71420 sequences (10000219 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32109, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.47)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71432 reads in 5.904 CPU sec, 5.834 real sec\n",
            "[M::process] read 71426 sequences (10000173 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32043, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
            "[M::mem_pestat] mean and std.dev: (340.11, 28.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 5.947 CPU sec, 5.876 real sec\n",
            "[M::process] read 71452 sequences (10000147 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31987, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.03, 28.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 5.923 CPU sec, 5.855 real sec\n",
            "[M::process] read 71436 sequences (10000127 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32041, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.88, 28.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71452 reads in 6.016 CPU sec, 5.951 real sec\n",
            "[M::process] read 71410 sequences (10000232 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32120, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.79, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71436 reads in 6.046 CPU sec, 5.972 real sec\n",
            "[M::process] read 71428 sequences (10000217 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32047, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 6.023 CPU sec, 5.952 real sec\n",
            "[M::process] read 71458 sequences (10000154 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32075, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.84, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 6.086 CPU sec, 6.021 real sec\n",
            "[M::process] read 71372 sequences (10000036 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32102, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.82, 28.27)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71458 reads in 6.007 CPU sec, 5.942 real sec\n",
            "[M::process] read 71448 sequences (10000152 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 32032, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71372 reads in 6.107 CPU sec, 6.041 real sec\n",
            "[M::process] read 71420 sequences (10000131 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 31993, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.27, 28.47)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71448 reads in 6.388 CPU sec, 6.330 real sec\n",
            "[M::process] read 71410 sequences (10000239 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34703, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 6.312 CPU sec, 6.247 real sec\n",
            "[M::process] read 71386 sequences (10000174 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34668, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.83, 28.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 6.510 CPU sec, 6.451 real sec\n",
            "[M::process] read 71438 sequences (10000243 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34675, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.96, 28.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71386 reads in 6.740 CPU sec, 6.672 real sec\n",
            "[M::process] read 71450 sequences (10000104 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34713, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.96, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 6.758 CPU sec, 6.686 real sec\n",
            "[M::process] read 71446 sequences (10000209 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34652, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.94, 28.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71450 reads in 6.750 CPU sec, 6.667 real sec\n",
            "[M::process] read 71468 sequences (10000136 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34661, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.91, 28.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71446 reads in 6.762 CPU sec, 6.680 real sec\n",
            "[M::process] read 71426 sequences (10000153 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34659, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.76, 28.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71468 reads in 6.815 CPU sec, 6.743 real sec\n",
            "[M::process] read 71434 sequences (10000004 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34677, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.97, 28.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 6.617 CPU sec, 6.543 real sec\n",
            "[M::process] read 71454 sequences (10000006 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34762, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.67, 28.39)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71434 reads in 6.633 CPU sec, 6.568 real sec\n",
            "[M::process] read 71428 sequences (10000092 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34663, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
            "[M::mem_pestat] mean and std.dev: (340.09, 28.06)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71454 reads in 6.659 CPU sec, 6.599 real sec\n",
            "[M::process] read 71398 sequences (10000311 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34668, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.90, 28.29)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 6.666 CPU sec, 6.593 real sec\n",
            "[M::process] read 71420 sequences (10000117 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34657, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71398 reads in 6.624 CPU sec, 6.549 real sec\n",
            "[M::process] read 71444 sequences (10000211 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34622, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.04, 28.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 6.628 CPU sec, 6.548 real sec\n",
            "[M::process] read 71432 sequences (10000233 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34706, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.03, 28.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71444 reads in 6.572 CPU sec, 6.505 real sec\n",
            "[M::process] read 71428 sequences (10000103 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34732, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.94, 28.53)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71432 reads in 6.600 CPU sec, 6.527 real sec\n",
            "[M::process] read 71472 sequences (10000188 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34687, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.82, 28.24)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 6.590 CPU sec, 6.508 real sec\n",
            "[M::process] read 71436 sequences (10000190 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34751, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.69, 28.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71472 reads in 6.530 CPU sec, 6.457 real sec\n",
            "[M::process] read 71412 sequences (10000256 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34680, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.75, 28.34)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71436 reads in 6.557 CPU sec, 6.481 real sec\n",
            "[M::process] read 71424 sequences (10000085 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34659, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.64, 28.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71412 reads in 6.637 CPU sec, 6.558 real sec\n",
            "[M::process] read 71448 sequences (10000016 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34675, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.08)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71424 reads in 6.681 CPU sec, 6.604 real sec\n",
            "[M::process] read 71434 sequences (10000063 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34679, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.62, 28.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71448 reads in 6.680 CPU sec, 6.600 real sec\n",
            "[M::process] read 71438 sequences (10000125 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.80, 28.42)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71434 reads in 6.586 CPU sec, 6.505 real sec\n",
            "[M::process] read 71430 sequences (10000063 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34724, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.97, 28.31)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 6.511 CPU sec, 6.442 real sec\n",
            "[M::process] read 71406 sequences (10000161 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34668, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.97, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 6.571 CPU sec, 6.492 real sec\n",
            "[M::process] read 71396 sequences (10000137 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34667, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.86, 28.38)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71406 reads in 6.581 CPU sec, 6.506 real sec\n",
            "[M::process] read 71438 sequences (10000268 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34695, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.00, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71396 reads in 6.660 CPU sec, 6.581 real sec\n",
            "[M::process] read 71446 sequences (10000051 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34705, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.71, 28.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 6.606 CPU sec, 6.535 real sec\n",
            "[M::process] read 71410 sequences (10000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34633, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.53, 28.19)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71446 reads in 6.615 CPU sec, 6.542 real sec\n",
            "[M::process] read 71440 sequences (10000247 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34612, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (339.93, 28.25)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71410 reads in 6.618 CPU sec, 6.538 real sec\n",
            "[M::process] read 71448 sequences (10000172 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34713, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.91, 28.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71440 reads in 6.614 CPU sec, 6.542 real sec\n",
            "[M::process] read 71438 sequences (10000136 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34692, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.91, 28.18)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71448 reads in 6.627 CPU sec, 6.543 real sec\n",
            "[M::process] read 71426 sequences (10000184 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.05, 28.28)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 6.611 CPU sec, 6.533 real sec\n",
            "[M::process] read 71466 sequences (10000144 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34700, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.04, 28.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 6.543 CPU sec, 6.472 real sec\n",
            "[M::process] read 71412 sequences (10000173 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34701, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.69, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71466 reads in 6.544 CPU sec, 6.467 real sec\n",
            "[M::process] read 71426 sequences (10000012 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34630, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
            "[M::mem_pestat] mean and std.dev: (340.17, 28.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71412 reads in 6.600 CPU sec, 6.523 real sec\n",
            "[M::process] read 71442 sequences (10000269 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34682, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.76, 28.33)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 6.643 CPU sec, 6.570 real sec\n",
            "[M::process] read 71426 sequences (10000171 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34672, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.88, 28.17)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71442 reads in 6.649 CPU sec, 6.576 real sec\n",
            "[M::process] read 71420 sequences (10000020 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34645, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.83, 28.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 6.699 CPU sec, 6.628 real sec\n",
            "[M::process] read 71428 sequences (10000030 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34682, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.91, 28.23)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71420 reads in 6.617 CPU sec, 6.540 real sec\n",
            "[M::process] read 71416 sequences (10000210 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34693, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.94, 28.37)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 6.653 CPU sec, 6.574 real sec\n",
            "[M::process] read 71452 sequences (10000132 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34682, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.98, 28.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71416 reads in 6.747 CPU sec, 6.679 real sec\n",
            "[M::process] read 71428 sequences (10000005 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34677, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.81, 28.26)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71452 reads in 6.654 CPU sec, 6.579 real sec\n",
            "[M::process] read 71414 sequences (10000143 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34732, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.64, 28.11)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71428 reads in 6.606 CPU sec, 6.533 real sec\n",
            "[M::process] read 71422 sequences (10000047 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34645, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.94, 28.30)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71414 reads in 6.727 CPU sec, 6.651 real sec\n",
            "[M::process] read 71408 sequences (10000026 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34662, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.72, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71422 reads in 6.636 CPU sec, 6.561 real sec\n",
            "[M::process] read 71424 sequences (10000102 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34637, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.77, 28.22)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71408 reads in 6.804 CPU sec, 6.734 real sec\n",
            "[M::process] read 71472 sequences (10000002 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34735, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.72, 28.32)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71424 reads in 6.660 CPU sec, 6.590 real sec\n",
            "[M::process] read 71438 sequences (10000196 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34719, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.44)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71472 reads in 6.658 CPU sec, 6.595 real sec\n",
            "[M::process] read 71426 sequences (10000161 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34685, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.87, 28.14)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 6.600 CPU sec, 6.529 real sec\n",
            "[M::process] read 71446 sequences (10000243 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34728, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.82, 28.15)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71426 reads in 6.646 CPU sec, 6.574 real sec\n",
            "[M::process] read 71442 sequences (10000007 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34651, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
            "[M::mem_pestat] mean and std.dev: (339.78, 27.98)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71446 reads in 6.735 CPU sec, 6.665 real sec\n",
            "[M::process] read 71438 sequences (10000251 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34696, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.91, 28.20)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71442 reads in 6.633 CPU sec, 6.558 real sec\n",
            "[M::process] read 71430 sequences (10000233 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34621, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.06, 28.35)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 6.707 CPU sec, 6.641 real sec\n",
            "[M::process] read 71438 sequences (10000245 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34668, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.95, 28.21)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71430 reads in 6.634 CPU sec, 6.564 real sec\n",
            "[M::process] read 71086 sequences (9953174 bp)...\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34736, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (340.02, 28.07)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71438 reads in 6.600 CPU sec, 6.531 real sec\n",
            "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34536, 0, 0)\n",
            "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
            "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
            "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
            "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
            "[M::mem_pestat] mean and std.dev: (339.78, 28.36)\n",
            "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
            "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
            "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
            "[M::mem_process_seqs] Processed 71086 reads in 6.519 CPU sec, 6.537 real sec\n",
            "[main] Version: 0.7.18-r1243-dirty\n",
            "[main] CMD: bwa mem -R @RG\\tID:sample1\\tSM:sample1\\tPL:ILLUMINA\\tLB:lib1\\tPU:unit1 reference/Chr7-reference.fasta cutadapt/caso3-tumor-pulmonar-trimmed_R1.fastq.gz cutadapt/caso3-tumor-pulmonar-trimmed_R2.fastq.gz\n",
            "[main] Real time: 689.741 sec; CPU: 697.756 sec\n",
            "[bam_sort_core] merging from 3 files and 1 in-memory blocks...\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "#1o indexa contra o genoma humano pra depois alinhar\n",
        "samtools faidx reference/Chr7-reference.fasta\n",
        "bwa index reference/Chr7-reference.fasta\n",
        "\n",
        "bwa mem reference/Chr7-reference.fasta \\\n",
        "  -R '@RG\\tID:sample1\\tSM:sample1\\tPL:ILLUMINA\\tLB:lib1\\tPU:unit1' \\\n",
        "  cutadapt/caso3-tumor-pulmonar-trimmed_R1.fastq.gz \\\n",
        "  cutadapt/caso3-tumor-pulmonar-trimmed_R2.fastq.gz \\\n",
        "  | samtools sort -o bwa/caso3-tumor-pulmonar-mapped-host.bam\n",
        "  \n",
        "  samtools index bwa/caso3-tumor-pulmonar-mapped-host.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {},
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Using GATK jar /home/figa/Documents/Programming/VSCode/Metagenomics-Course/env/share/gatk4-4.6.2.0-1/gatk-package-4.6.2.0-local.jar\n",
            "Running:\n",
            "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /home/figa/Documents/Programming/VSCode/Metagenomics-Course/env/share/gatk4-4.6.2.0-1/gatk-package-4.6.2.0-local.jar Mutect2 -R reference/Chr7-reference.fasta -I bwa/caso3-tumor-pulmonar-mapped-host.bam -O mutect/caso3-tumor-pulmonar-unfiletered.vcf\n",
            "09:36:50.265 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/home/figa/Documents/Programming/VSCode/Metagenomics-Course/env/share/gatk4-4.6.2.0-1/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "09:36:50.313 INFO  Mutect2 - ------------------------------------------------------------\n",
            "09:36:50.314 INFO  Mutect2 - The Genome Analysis Toolkit (GATK) v4.6.2.0\n",
            "09:36:50.314 INFO  Mutect2 - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
            "09:36:50.314 INFO  Mutect2 - Executing as figa@FIGUEIREOS on Linux v6.17.1-arch1-1 amd64\n",
            "09:36:50.314 INFO  Mutect2 - Java runtime: OpenJDK 64-Bit Server VM v22.0.1-internal-adhoc.conda.src\n",
            "09:36:50.314 INFO  Mutect2 - Start Date/Time: October 13, 2025, 9:36:50â¯AM BRT\n",
            "09:36:50.314 INFO  Mutect2 - ------------------------------------------------------------\n",
            "09:36:50.314 INFO  Mutect2 - ------------------------------------------------------------\n",
            "09:36:50.315 INFO  Mutect2 - HTSJDK Version: 4.2.0\n",
            "09:36:50.315 INFO  Mutect2 - Picard Version: 3.4.0\n",
            "09:36:50.315 INFO  Mutect2 - Built for Spark Version: 3.5.0\n",
            "09:36:50.316 INFO  Mutect2 - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
            "09:36:50.316 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
            "09:36:50.316 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
            "09:36:50.316 INFO  Mutect2 - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
            "09:36:50.316 INFO  Mutect2 - Deflater: IntelDeflater\n",
            "09:36:50.316 INFO  Mutect2 - Inflater: IntelInflater\n",
            "09:36:50.316 INFO  Mutect2 - GCS max retries/reopens: 20\n",
            "09:36:50.316 INFO  Mutect2 - Requester pays: disabled\n",
            "09:36:50.316 INFO  Mutect2 - Initializing engine\n",
            "09:36:50.372 INFO  Mutect2 - Done initializing engine\n",
            "09:36:50.376 INFO  NativeLibraryLoader - Loading libgkl_utils.so from jar:file:/home/figa/Documents/Programming/VSCode/Metagenomics-Course/env/share/gatk4-4.6.2.0-1/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_utils.so\n",
            "09:36:50.377 INFO  NativeLibraryLoader - Loading libgkl_smithwaterman.so from jar:file:/home/figa/Documents/Programming/VSCode/Metagenomics-Course/env/share/gatk4-4.6.2.0-1/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_smithwaterman.so\n",
            "09:36:50.377 INFO  IntelSmithWaterman - Using CPU-supported AVX-512 instructions\n",
            "09:36:50.377 INFO  SmithWatermanAligner - Using AVX accelerated SmithWaterman implementation\n",
            "09:36:50.391 INFO  NativeLibraryLoader - Loading libgkl_pairhmm_omp.so from jar:file:/home/figa/Documents/Programming/VSCode/Metagenomics-Course/env/share/gatk4-4.6.2.0-1/gatk-package-4.6.2.0-local.jar!/com/intel/gkl/native/libgkl_pairhmm_omp.so\n",
            "09:36:50.396 INFO  IntelPairHmm - Using CPU-supported AVX-512 instructions\n",
            "09:36:50.396 INFO  IntelPairHmm - Flush-to-zero (FTZ) is enabled when running PairHMM\n",
            "09:36:50.397 INFO  IntelPairHmm - Available threads: 16\n",
            "09:36:50.397 INFO  IntelPairHmm - Requested threads: 4\n",
            "09:36:50.397 INFO  PairHMM - Using the OpenMP multi-threaded AVX-accelerated native PairHMM implementation\n",
            "09:36:50.413 INFO  ProgressMeter - Starting traversal\n",
            "09:36:50.414 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Regions Processed   Regions/Minute\n",
            "09:37:00.432 INFO  ProgressMeter - NC_000007.14:54626562              0.2                182100        1090745.7\n",
            "09:37:10.468 INFO  ProgressMeter - NC_000007.14:55267302              0.3                184250         551289.1\n",
            "09:37:20.507 INFO  ProgressMeter - NC_000007.14:55914597              0.5                186420         371687.8\n",
            "09:37:30.510 INFO  ProgressMeter - NC_000007.14:139819516              0.7                466110         697491.0\n",
            "09:37:40.532 INFO  ProgressMeter - NC_000007.14:140470740              0.8                468290         560636.1\n",
            "09:37:50.536 INFO  ProgressMeter - NC_000007.14:141112381              1.0                470440         469485.4\n",
            "09:38:00.578 INFO  ProgressMeter - NC_000007.14:141751099              1.2                472580         404121.8\n",
            "09:38:01.792 INFO  Mutect2 - 44886 read(s) filtered by: MappingQualityReadFilter \n",
            "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
            "0 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
            "0 read(s) filtered by: MappedReadFilter \n",
            "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
            "0 read(s) filtered by: NotDuplicateReadFilter \n",
            "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
            "0 read(s) filtered by: NonChimericOriginalAlignmentReadFilter \n",
            "0 read(s) filtered by: NonZeroReferenceLengthAlignmentReadFilter \n",
            "0 read(s) filtered by: ReadLengthReadFilter \n",
            "0 read(s) filtered by: GoodCigarReadFilter \n",
            "0 read(s) filtered by: WellformedReadFilter \n",
            "44886 total reads filtered out of 7999998 reads processed\n",
            "09:38:01.792 INFO  ProgressMeter - NC_000007.14:159343099              1.2                531229         446548.5\n",
            "09:38:01.792 INFO  ProgressMeter - Traversal complete. Processed 531229 total regions in 1.2 minutes.\n",
            "09:38:01.819 INFO  VectorLoglessPairHMM - Time spent in setup for JNI call : 5.05493E-4\n",
            "09:38:01.820 INFO  PairHMM - Total compute time in PairHMM computeLogLikelihoods() : 0.0015275870000000002\n",
            "09:38:01.820 INFO  SmithWatermanAligner - Total compute time in native Smith-Waterman : 0.00 sec\n",
            "09:38:01.820 INFO  Mutect2 - Shutting down engine\n",
            "[October 13, 2025, 9:38:01â¯AM BRT] org.broadinstitute.hellbender.tools.walkers.mutect.Mutect2 done. Elapsed time: 1.19 minutes.\n",
            "Runtime.totalMemory()=754974720\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Tool returned:\n",
            "SUCCESS\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Indexar a referÃªncia e BAM para chamada de variantes com GATK\n",
        "samtools dict reference/Chr7-reference.fasta > reference/Chr7-reference.dict\n",
        "samtools index bwa/caso3-tumor-pulmonar-mapped-host.bam\n",
        "\n",
        " gatk Mutect2 \\\n",
        " -R reference/Chr7-reference.fasta \\\n",
        "    -I bwa/caso3-tumor-pulmonar-mapped-host.bam \\\n",
        "    -O mutect/caso3-tumor-pulmonar-unfiletered.vcf"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# AnotaÃ§Ã£o de Variantes"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            " - getting list of available cache files\n",
            " - downloading https://ftp.ensembl.org/pub/release-115/variation/indexed_vep_cache/homo_sapiens_vep_115_GRCh38.tar.gz\n",
            " - unpacking homo_sapiens_vep_115_GRCh38.tar.gz\n",
            " - converting cache, this may take some time but will allow VEP to look up variants and frequency data much faster\n",
            " - use CTRL-C to cancel if you do not wish to convert this cache now (you may run convert_cache.pl later)\n",
            "2025-10-12 12:37:45 - Processing homo_sapiens\n",
            "2025-10-12 12:37:45 - Processing version 115_GRCh38\n",
            "2025-10-12 12:37:45 - No unprocessed types remaining, skipping\n",
            "2025-10-12 12:37:45 - All done!\n",
            " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz\n",
            " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz.fai\n",
            " - downloading Homo_sapiens.GRCh38.dna.toplevel.fa.gz.gzi\n",
            "\n",
            "The FASTA file should be automatically detected by the VEP when using --cache or --offline.\n",
            "If it is not, use \"--fasta /home/figa/.vep/homo_sapiens/115_GRCh38/Homo_sapiens.GRCh38.dna.toplevel.fa.gz\"\n",
            "\n",
            "\n",
            "All done\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "vep_install -a cf -s homo_sapiens -y GRCh38 --NO_UPDATE ##baixa o genoma humano\n",
        "##Oh fuck."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [],
      "source": [
        "%%bash\n",
        "bgzip gatk/caso3-tumor-pulmonar-host-variants.vcf\n",
        "tabix -p vcf gatk/caso3-tumor-pulmonar-host-variants.vcf.gz\n",
        "#gera os arquivos VCF para char o gene correto"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {},
      "outputs": [],
      "source": [
        "%%bash\n",
        "# AnotaÃ§Ã£o de variantes a nÃ­vel de proteÃ­na\n",
        "vep -i gatk/caso3-tumor-pulmonar-host-variants.vcf.gz \\\n",
        "    --output_file vep/caso3-tumor-pulmonar-host-variants.vep.tsv \\\n",
        "    --tab --force_overwrite \\\n",
        "    --offline --cache --dir_cache ~/.vep \\\n",
        "    --assembly GRCh38 \\\n",
        "    --symbol --hgvs --canonical --af_1kg --max_af \\\n",
        "    --everything --pick\\\n",
        "    --stats_file vep/caso3-tumor-pulmonar-host-variants.vep.html"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# AnÃ¡lise e InterpretaÃ§Ã£o dos Resultados"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Total de variantes anotadas: 2\n",
            "\n",
            "Colunas (primeiras 40):\n",
            "['Uploaded_variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature_type', 'Consequence', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL', 'SYMBOL_SOURCE', 'HGNC_ID', 'BIOTYPE', 'CANONICAL', 'MANE', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS']\n"
          ]
        },
        {
          "data": {
            "application/vnd.microsoft.datawrangler.viewer.v0+json": {
              "columns": [
                {
                  "name": "index",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "Uploaded_variation",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Location",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Allele",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Gene",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Feature",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Feature_type",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Consequence",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "cDNA_position",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "CDS_position",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "Protein_position",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "Amino_acids",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Codons",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Existing_variation",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "IMPACT",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "DISTANCE",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "STRAND",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "FLAGS",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "VARIANT_CLASS",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "SYMBOL",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "SYMBOL_SOURCE",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "HGNC_ID",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "BIOTYPE",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "CANONICAL",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MANE",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MANE_SELECT",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MANE_PLUS_CLINICAL",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "TSL",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "APPRIS",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "CCDS",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "ENSP",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "SWISSPROT",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "TREMBL",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "UNIPARC",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "UNIPROT_ISOFORM",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "GENE_PHENO",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "SIFT",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "PolyPhen",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "EXON",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "INTRON",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "DOMAINS",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "miRNA",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "HGVSc",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "HGVSp",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "HGVS_OFFSET",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "AFR_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "AMR_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "EAS_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "EUR_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "SAS_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_AFR_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_AMR_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_ASJ_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_EAS_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_FIN_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_MID_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_NFE_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_REMAINING_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADe_SAS_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_AFR_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_AMI_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_AMR_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_ASJ_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_EAS_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_FIN_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_MID_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_NFE_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_REMAINING_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "gnomADg_SAS_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MAX_AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MAX_AF_POPS",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "CLIN_SIG",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "SOMATIC",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "PHENO",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "PUBMED",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MOTIF_NAME",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MOTIF_POS",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "HIGH_INF_POS",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MOTIF_SCORE_CHANGE",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "TRANSCRIPTION_FACTORS",
                  "rawType": "object",
                  "type": "string"
                }
              ],
              "ref": "13cd2942-1688-47e6-9dd3-7e3c15b0c306",
              "rows": [
                [
                  "0",
                  "NC_000007.14_55191822_T/G",
                  "NC_000007.14:55191822",
                  "G",
                  "ENSG00000146648",
                  "ENST00000275493",
                  "Transcript",
                  "missense_variant",
                  "2834",
                  "2573",
                  "858",
                  "L/R",
                  "cTg/cGg",
                  "rs121434568,COSV51765161,COSV51854030",
                  "MODERATE",
                  "-",
                  "1",
                  "-",
                  "SNV",
                  "EGFR",
                  "HGNC",
                  "HGNC:3236",
                  "protein_coding",
                  "YES",
                  "MANE_Select",
                  "NM_005228.5",
                  "-",
                  "1",
                  "P3",
                  "CCDS5514.1",
                  "ENSP00000275493",
                  "P00533.296",
                  "-",
                  "UPI000003E750",
                  "P00533-1",
                  "1",
                  "deleterious_low_confidence(0)",
                  "probably_damaging(0.997)",
                  "21/28",
                  "-",
                  "Gene3D:1.10.510.10,PDB-ENSP_mappings:1m14.A,PDB-ENSP_mappings:1m17.A,PDB-ENSP_mappings:1xkk.A,PDB-ENSP_mappings:2eb2.A,PDB-ENSP_mappings:2eb3.A,PDB-ENSP_mappings:2gs2.A,PDB-ENSP_mappings:2gs6.A,PDB-ENSP_mappings:2gs7.A,PDB-ENSP_mappings:2gs7.B,PDB-ENSP_mappings:2itn.A,PDB-ENSP_mappings:2ito.A,PDB-ENSP_mappings:2itp.A,PDB-ENSP_mappings:2itq.A,PDB-ENSP_mappings:2itt.A,PDB-ENSP_mappings:2itu.A,PDB-ENSP_mappings:2itv.A,PDB-ENSP_mappings:2itw.A,PDB-ENSP_mappings:2itx.A,PDB-ENSP_mappings:2ity.A,PDB-ENSP_mappings:2itz.A,PDB-ENSP_mappings:2j5e.A,PDB-ENSP_mappings:2j5f.A,PDB-ENSP_mappings:2j6m.A,PDB-ENSP_mappings:2jit.A,PDB-ENSP_mappings:2jit.B,PDB-ENSP_mappings:2jiu.A,PDB-ENSP_mappings:2jiu.B,PDB-ENSP_mappings:2jiv.A,PDB-ENSP_mappings:2jiv.B,PDB-ENSP_mappings:2rf9.A,PDB-ENSP_mappings:2rf9.B,PDB-ENSP_mappings:2rfd.A,PDB-ENSP_mappings:2rfd.B,PDB-ENSP_mappings:2rfe.A,PDB-ENSP_mappings:2rfe.B,PDB-ENSP_mappings:2rfe.C,PDB-ENSP_mappings:2rfe.D,PDB-ENSP_mappings:2rgp.A,PDB-ENSP_mappings:3bel.A,PDB-ENSP_mappings:3gop.A,PDB-ENSP_mappings:3gt8.A,PDB-ENSP_mappings:3gt8.B,PDB-ENSP_mappings:3gt8.C,PDB-ENSP_mappings:3gt8.D,PDB-ENSP_mappings:3ika.A,PDB-ENSP_mappings:3ika.B,PDB-ENSP_mappings:3lzb.A,PDB-ENSP_mappings:3lzb.B,PDB-ENSP_mappings:3lzb.C,PDB-ENSP_mappings:3lzb.D,PDB-ENSP_mappings:3lzb.E,PDB-ENSP_mappings:3lzb.F,PDB-ENSP_mappings:3lzb.G,PDB-ENSP_mappings:3lzb.H,PDB-ENSP_mappings:3poz.A,PDB-ENSP_mappings:3ug1.A,PDB-ENSP_mappings:3ug2.A,PDB-ENSP_mappings:3vjn.A,PDB-ENSP_mappings:3vjo.A,PDB-ENSP_mappings:3w2o.A,PDB-ENSP_mappings:3w2p.A,PDB-ENSP_mappings:3w2q.A,PDB-ENSP_mappings:3w2r.A,PDB-ENSP_mappings:3w2s.A,PDB-ENSP_mappings:3w32.A,PDB-ENSP_mappings:3w33.A,PDB-ENSP_mappings:4g5j.A,PDB-ENSP_mappings:4g5p.A,PDB-ENSP_mappings:4g5p.B,PDB-ENSP_mappings:4hjo.A,PDB-ENSP_mappings:4i1z.A,PDB-ENSP_mappings:4i20.A,PDB-ENSP_mappings:4i21.A,PDB-ENSP_mappings:4i21.B,PDB-ENSP_mappings:4i22.A,PDB-ENSP_mappings:4i23.A,PDB-ENSP_mappings:4i24.A,PDB-ENSP_mappings:4i24.B,PDB-ENSP_mappings:4jq7.A,PDB-ENSP_mappings:4jq8.A,PDB-ENSP_mappings:4jr3.A,PDB-ENSP_mappings:4jrv.A,PDB-ENSP_mappings:4li5.A,PDB-ENSP_mappings:4ll0.A,PDB-ENSP_mappings:4ll0.B,PDB-ENSP_mappings:4lqm.A,PDB-ENSP_mappings:4lrm.A,PDB-ENSP_mappings:4lrm.B,PDB-ENSP_mappings:4lrm.C,PDB-ENSP_mappings:4lrm.D,PDB-ENSP_mappings:4lrm.E,PDB-ENSP_mappings:4r3p.A,PDB-ENSP_mappings:4r3r.A,PDB-ENSP_mappings:4r5s.A,PDB-ENSP_mappings:4riw.B,PDB-ENSP_mappings:4riw.D,PDB-ENSP_mappings:4rix.B,PDB-ENSP_mappings:4rix.D,PDB-ENSP_mappings:4riy.B,PDB-ENSP_mappings:4riy.D,PDB-ENSP_mappings:4rj4.A,PDB-ENSP_mappings:4rj5.A,PDB-ENSP_mappings:4rj6.A,PDB-ENSP_mappings:4rj7.A,PDB-ENSP_mappings:4rj8.A,PDB-ENSP_mappings:4tks.A,PDB-ENSP_mappings:4wd5.A,PDB-ENSP_mappings:4wd5.B,PDB-ENSP_mappings:4wkq.A,PDB-ENSP_mappings:4wrg.A,PDB-ENSP_mappings:4zau.A,PDB-ENSP_mappings:4zjv.A,PDB-ENSP_mappings:4zjv.B,PDB-ENSP_mappings:4zse.A,PDB-ENSP_mappings:4zse.B,PDB-ENSP_mappings:4zse.C,PDB-ENSP_mappings:4zse.D,PDB-ENSP_mappings:5c8k.A,PDB-ENSP_mappings:5c8m.A,PDB-ENSP_mappings:5c8n.A,PDB-ENSP_mappings:5cal.A,PDB-ENSP_mappings:5can.A,PDB-ENSP_mappings:5cao.A,PDB-ENSP_mappings:5cap.A,PDB-ENSP_mappings:5caq.A,PDB-ENSP_mappings:5cas.A,PDB-ENSP_mappings:5cau.A,PDB-ENSP_mappings:5cav.A,PDB-ENSP_mappings:5cnn.A,PDB-ENSP_mappings:5cnn.B,PDB-ENSP_mappings:5cno.A,PDB-ENSP_mappings:5cno.B,PDB-ENSP_mappings:5cno.X,PDB-ENSP_mappings:5czh.A,PDB-ENSP_mappings:5czi.A,PDB-ENSP_mappings:5d41.A,PDB-ENSP_mappings:5d41.B,PDB-ENSP_mappings:5edp.A,PDB-ENSP_mappings:5edq.A,PDB-ENSP_mappings:5edr.A,PDB-ENSP_mappings:5em5.A,PDB-ENSP_mappings:5em6.A,PDB-ENSP_mappings:5em7.A,PDB-ENSP_mappings:5em8.A,PDB-ENSP_mappings:5fed.A,PDB-ENSP_mappings:5fee.A,PDB-ENSP_mappings:5feq.A,PDB-ENSP_mappings:5gmp.A,PDB-ENSP_mappings:5gnk.A,PDB-ENSP_mappings:5gty.A,PDB-ENSP_mappings:5gty.B,PDB-ENSP_mappings:5gty.C,PDB-ENSP_mappings:5gty.D,PDB-ENSP_mappings:5gty.E,PDB-ENSP_mappings:5gty.F,PDB-ENSP_mappings:5gty.G,PDB-ENSP_mappings:5gty.H,PDB-ENSP_mappings:5gtz.A,PDB-ENSP_mappings:5hcx.A,PDB-ENSP_mappings:5hcy.A,PDB-ENSP_mappings:5hcz.A,PDB-ENSP_mappings:5hg5.A,PDB-ENSP_mappings:5hg7.A,PDB-ENSP_mappings:5hg8.A,PDB-ENSP_mappings:5hg9.A,PDB-ENSP_mappings:5hib.A,PDB-ENSP_mappings:5hic.A,PDB-ENSP_mappings:5j9y.A,PDB-ENSP_mappings:5j9z.A,PDB-ENSP_mappings:5jeb.A,PDB-ENSP_mappings:5u8l.A,PDB-ENSP_mappings:5ug8.A,PDB-ENSP_mappings:5ug9.A,PDB-ENSP_mappings:5uga.A,PDB-ENSP_mappings:5ugb.A,PDB-ENSP_mappings:5ugc.A,PDB-ENSP_mappings:5uwd.A,PDB-ENSP_mappings:5x26.A,PDB-ENSP_mappings:5x27.A,PDB-ENSP_mappings:5x28.A,PDB-ENSP_mappings:5x2a.A,PDB-ENSP_mappings:5x2a.B,PDB-ENSP_mappings:5x2a.C,PDB-ENSP_mappings:5x2a.D,PDB-ENSP_mappings:5x2c.A,PDB-ENSP_mappings:5x2c.B,PDB-ENSP_mappings:5x2f.A,PDB-ENSP_mappings:5x2f.B,PDB-ENSP_mappings:5x2f.C,PDB-ENSP_mappings:5x2f.D,PDB-ENSP_mappings:5x2k.A,PDB-ENSP_mappings:5xdk.A,PDB-ENSP_mappings:5xdl.A,PDB-ENSP_mappings:5xgm.A,PDB-ENSP_mappings:5xgn.A,PDB-ENSP_mappings:5xgn.B,PDB-ENSP_mappings:5y25.A,PDB-ENSP_mappings:5y9t.A,PDB-ENSP_mappings:5yu9.A,PDB-ENSP_mappings:5yu9.B,PDB-ENSP_mappings:5yu9.C,PDB-ENSP_mappings:5yu9.D,PDB-ENSP_mappings:5zto.A,PDB-ENSP_mappings:5zwj.A,PDB-ENSP_mappings:6d8e.A,PDB-ENSP_mappings:6duk.A,PDB-ENSP_mappings:6duk.B,PDB-ENSP_mappings:6duk.C,PDB-ENSP_mappings:6duk.D,PDB-ENSP_mappings:6duk.E,PDB-ENSP_mappings:6duk.F,PDB-ENSP_mappings:6jrj.A,PDB-ENSP_mappings:6jrk.A,PDB-ENSP_mappings:6jrx.A,PDB-ENSP_mappings:6jwl.A,PDB-ENSP_mappings:6jx0.A,PDB-ENSP_mappings:6jx4.A,PDB-ENSP_mappings:6jxt.A,PDB-ENSP_mappings:6jz0.A,PDB-ENSP_mappings:6lub.A,PDB-ENSP_mappings:6lud.A,PDB-ENSP_mappings:6p1d.A,PDB-ENSP_mappings:6p1d.B,PDB-ENSP_mappings:6p1d.C,PDB-ENSP_mappings:6p1d.D,PDB-ENSP_mappings:6p1l.A,PDB-ENSP_mappings:6p1l.B,PDB-ENSP_mappings:6p1l.C,PDB-ENSP_mappings:6p1l.D,PDB-ENSP_mappings:6p8q.A,PDB-ENSP_mappings:6p8q.B,PDB-ENSP_mappings:6p8q.C,PDB-ENSP_mappings:6p8q.D,PDB-ENSP_mappings:6s89.A,PDB-ENSP_mappings:6s8a.A,PDB-ENSP_mappings:6s9b.A,PDB-ENSP_mappings:6s9c.A,PDB-ENSP_mappings:6s9d.A,PDB-ENSP_mappings:6tfu.A,PDB-ENSP_mappings:6tfu.B,PDB-ENSP_mappings:6tfv.A,PDB-ENSP_mappings:6tfv.B,PDB-ENSP_mappings:6tfw.A,PDB-ENSP_mappings:6tfw.B,PDB-ENSP_mappings:6tfy.A,PDB-ENSP_mappings:6tfy.B,PDB-ENSP_mappings:6tfz.A,PDB-ENSP_mappings:6tfz.B,PDB-ENSP_mappings:6tg0.A,PDB-ENSP_mappings:6tg0.B,PDB-ENSP_mappings:6tg1.A,PDB-ENSP_mappings:6tg1.B,PDB-ENSP_mappings:6v5n.A,PDB-ENSP_mappings:6v5n.B,PDB-ENSP_mappings:6v5n.C,PDB-ENSP_mappings:6v5n.D,PDB-ENSP_mappings:6v5p.A,PDB-ENSP_mappings:6v5p.B,PDB-ENSP_mappings:6v5p.C,PDB-ENSP_mappings:6v5p.D,PDB-ENSP_mappings:6v66.A,PDB-ENSP_mappings:6v66.B,PDB-ENSP_mappings:6v66.C,PDB-ENSP_mappings:6v66.D,PDB-ENSP_mappings:6v6k.A,PDB-ENSP_mappings:6v6k.B,PDB-ENSP_mappings:6v6k.C,PDB-ENSP_mappings:6v6k.D,PDB-ENSP_mappings:6v6k.E,PDB-ENSP_mappings:6v6k.F,PDB-ENSP_mappings:6v6k.G,PDB-ENSP_mappings:6v6k.H,PDB-ENSP_mappings:6v6o.A,PDB-ENSP_mappings:6v6o.B,PDB-ENSP_mappings:6v6o.C,PDB-ENSP_mappings:6v6o.D,PDB-ENSP_mappings:6v6o.E,PDB-ENSP_mappings:6v6o.F,PDB-ENSP_mappings:6v6o.G,PDB-ENSP_mappings:6v6o.H,PDB-ENSP_mappings:6vh4.A,PDB-ENSP_mappings:6vhn.A,PDB-ENSP_mappings:6vhp.A,PDB-ENSP_mappings:6wa2.A,PDB-ENSP_mappings:6wa2.B,PDB-ENSP_mappings:6wa2.C,PDB-ENSP_mappings:6wa2.D,PDB-ENSP_mappings:6wak.A,PDB-ENSP_mappings:6wak.B,PDB-ENSP_mappings:6wak.C,PDB-ENSP_mappings:6wak.D,PDB-ENSP_mappings:6wxn.A,PDB-ENSP_mappings:6wxn.B,PDB-ENSP_mappings:6wxn.C,PDB-ENSP_mappings:6wxn.D,PDB-ENSP_mappings:6xl4.A,PDB-ENSP_mappings:6xl4.B,PDB-ENSP_mappings:6xl4.C,PDB-ENSP_mappings:6xl4.D,PDB-ENSP_mappings:6z4b.A,PDB-ENSP_mappings:6z4b.B,PDB-ENSP_mappings:6z4d.A,PDB-ENSP_mappings:6z4d.B,PDB-ENSP_mappings:7a2a.A,PDB-ENSP_mappings:7a2a.B,PDB-ENSP_mappings:7a6i.A,PDB-ENSP_mappings:7a6j.A,PDB-ENSP_mappings:7a6j.B,PDB-ENSP_mappings:7a6k.A,PDB-ENSP_mappings:7a6k.B,PDB-ENSP_mappings:7a6k.C,PDB-ENSP_mappings:7a6k.D,PDB-ENSP_mappings:7aei.A,PDB-ENSP_mappings:7aem.A,PDB-ENSP_mappings:7b85.A,PDB-ENSP_mappings:7er2.A,PDB-ENSP_mappings:7jxi.A,PDB-ENSP_mappings:7jxi.B,PDB-ENSP_mappings:7jxi.C,PDB-ENSP_mappings:7jxi.D,PDB-ENSP_mappings:7jxk.A,PDB-ENSP_mappings:7jxk.B,PDB-ENSP_mappings:7jxk.C,PDB-ENSP_mappings:7jxk.D,PDB-ENSP_mappings:7jxk.E,PDB-ENSP_mappings:7jxk.F,PDB-ENSP_mappings:7jxl.A,PDB-ENSP_mappings:7jxl.B,PDB-ENSP_mappings:7jxl.C,PDB-ENSP_mappings:7jxl.D,PDB-ENSP_mappings:7jxm.A,PDB-ENSP_mappings:7jxm.B,PDB-ENSP_mappings:7jxm.C,PDB-ENSP_mappings:7jxm.D,PDB-ENSP_mappings:7jxp.A,PDB-ENSP_mappings:7jxp.B,PDB-ENSP_mappings:7jxp.C,PDB-ENSP_mappings:7jxp.D,PDB-ENSP_mappings:7jxp.E,PDB-ENSP_mappings:7jxp.F,PDB-ENSP_mappings:7jxq.A,PDB-ENSP_mappings:7jxq.B,PDB-ENSP_mappings:7jxq.C,PDB-ENSP_mappings:7jxq.D,PDB-ENSP_mappings:7jxw.A,PDB-ENSP_mappings:7jxw.B,PDB-ENSP_mappings:7jxw.C,PDB-ENSP_mappings:7jxw.D,PDB-ENSP_mappings:7k1h.A,PDB-ENSP_mappings:7k1h.B,PDB-ENSP_mappings:7k1h.C,PDB-ENSP_mappings:7k1h.D,PDB-ENSP_mappings:7k1h.E,PDB-ENSP_mappings:7k1h.F,PDB-ENSP_mappings:7k1i.A,PDB-ENSP_mappings:7kxz.A,PDB-ENSP_mappings:7ky0.A,PDB-ENSP_mappings:7ky0.B,PDB-ENSP_mappings:7ky0.C,PDB-ENSP_mappings:7ky0.D,PDB-ENSP_mappings:7lg8.A,PDB-ENSP_mappings:7lg8.B,PDB-ENSP_mappings:7lg8.C,PDB-ENSP_mappings:7lg8.D,PDB-ENSP_mappings:7lgs.A,PDB-ENSP_mappings:7lgs.B,PDB-ENSP_mappings:7lgs.C,PDB-ENSP_mappings:7lgs.D,PDB-ENSP_mappings:7ltx.A,PDB-ENSP_mappings:7ltx.B,PDB-ENSP_mappings:7ltx.C,PDB-ENSP_mappings:7ltx.D,PDB-ENSP_mappings:7oxb.A,PDB-ENSP_mappings:7si1.A,PDB-ENSP_mappings:7syd.A,PDB-ENSP_mappings:7syd.B,PDB-ENSP_mappings:7sye.A,PDB-ENSP_mappings:7sye.B,PDB-ENSP_mappings:7sz0.A,PDB-ENSP_mappings:7sz0.B,PDB-ENSP_mappings:7sz1.A,PDB-ENSP_mappings:7sz1.B,PDB-ENSP_mappings:7sz5.A,PDB-ENSP_mappings:7sz5.B,PDB-ENSP_mappings:7sz7.A,PDB-ENSP_mappings:7sz7.B,PDB-ENSP_mappings:7t4i.A,PDB-ENSP_mappings:7t4j.A,PDB-ENSP_mappings:7t4j.B,PDB-ENSP_mappings:7tvd.A,PDB-ENSP_mappings:7u98.A,PDB-ENSP_mappings:7u98.B,PDB-ENSP_mappings:7u98.C,PDB-ENSP_mappings:7u98.D,PDB-ENSP_mappings:7u99.A,PDB-ENSP_mappings:7u9a.A,PDB-ENSP_mappings:7ukv.A,PDB-ENSP_mappings:7ukw.A,PDB-ENSP_mappings:7ukw.B,PDB-ENSP_mappings:7ukw.C,PDB-ENSP_mappings:7ukw.D,PDB-ENSP_mappings:7vra.A,PDB-ENSP_mappings:7vre.A,PDB-ENSP_mappings:7zym.A,PDB-ENSP_mappings:7zyn.A,PDB-ENSP_mappings:7zyp.A,PDB-ENSP_mappings:7zyq.A,PDB-ENSP_mappings:7zyq.B,PDB-ENSP_mappings:8a27.A,PDB-ENSP_mappings:8a2a.A,PDB-ENSP_mappings:8a2b.A,PDB-ENSP_mappings:8a2d.A,PDB-ENSP_mappings:8d73.A,PDB-ENSP_mappings:8d73.B,PDB-ENSP_mappings:8d76.A,PDB-ENSP_mappings:8d76.B,PDB-ENSP_mappings:8dsw.A,PDB-ENSP_mappings:8eme.A,PDB-ENSP_mappings:8eme.B,PDB-ENSP_mappings:8f1h.A,PDB-ENSP_mappings:8f1w.A,PDB-ENSP_mappings:8f1w.B,PDB-ENSP_mappings:8f1w.C,PDB-ENSP_mappings:8f1w.D,PDB-ENSP_mappings:8f1x.A,PDB-ENSP_mappings:8f1y.A,PDB-ENSP_mappings:8f1z.A,PDB-ENSP_mappings:8fv3.A,PDB-ENSP_mappings:8fv3.B,PDB-ENSP_mappings:8fv3.C,PDB-ENSP_mappings:8fv3.D,PDB-ENSP_mappings:8fv4.A,PDB-ENSP_mappings:8fv4.B,PDB-ENSP_mappings:8fv4.C,PDB-ENSP_mappings:8fv4.D,PDB-ENSP_mappings:8g63.A,PDB-ENSP_mappings:8gb4.A,PDB-ENSP_mappings:8gb4.B,PDB-ENSP_mappings:8gb4.C,PDB-ENSP_mappings:8gb4.D,PDB-ENSP_mappings:8gk5.A,PDB-ENSP_mappings:8gk5.B,PDB-ENSP_mappings:8gk5.C,PDB-ENSP_mappings:8gk5.D,PDB-ENSP_mappings:8h7x.A,PDB-ENSP_mappings:8h7x.B,PDB-ENSP_mappings:8hv1.A,PDB-ENSP_mappings:8hv2.A,PDB-ENSP_mappings:8hv3.A,PDB-ENSP_mappings:8hv4.A,PDB-ENSP_mappings:8hv5.A,PDB-ENSP_mappings:8hv6.A,PDB-ENSP_mappings:8hv7.A,PDB-ENSP_mappings:8hv8.A,PDB-ENSP_mappings:8hv9.A,PDB-ENSP_mappings:8hva.A,PDB-ENSP_mappings:8hy7.A,PDB-ENSP_mappings:8kfq.A,PDB-ENSP_mappings:8pnz.A,PDB-ENSP_mappings:8po0.A,PDB-ENSP_mappings:8po1.A,PDB-ENSP_mappings:8po2.A,PDB-ENSP_mappings:8po2.B,PDB-ENSP_mappings:8po2.D,PDB-ENSP_mappings:8po3.A,PDB-ENSP_mappings:8po4.A,PDB-ENSP_mappings:8po4.B,PDB-ENSP_mappings:8sc7.A,PDB-ENSP_mappings:8tjl.A,PDB-ENSP_mappings:8to3.A,PDB-ENSP_mappings:8to3.B,PDB-ENSP_mappings:8to3.C,PDB-ENSP_mappings:8to3.D,PDB-ENSP_mappings:8to4.A,PDB-ENSP_mappings:8to4.B,PDB-ENSP_mappings:8to4.C,PDB-ENSP_mappings:8to4.D,PDB-ENSP_mappings:8wd4.A,PDB-ENSP_mappings:9ews.A,PDB-ENSP_mappings:9ewt.A,PDB-ENSP_mappings:9ewt.B,PDB-ENSP_mappings:9fqp.A,PDB-ENSP_mappings:9fqs.A,PDB-ENSP_mappings:9frd.A,PDB-ENSP_mappings:9gc4.A,PDB-ENSP_mappings:9gc4.B,PDB-ENSP_mappings:9gc4.D,PDB-ENSP_mappings:9gc5.A,PDB-ENSP_mappings:9gc5.B,PDB-ENSP_mappings:9gc6.A,PDB-ENSP_mappings:9gc6.B,PDB-ENSP_mappings:9gdv.A,PDB-ENSP_mappings:9gdv.B,PDB-ENSP_mappings:9h46.A,PDB-ENSP_mappings:9hbo.A,PDB-ENSP_mappings:9hbo.B,AFDB-ENSP_mappings:AF-P00533-F1,Phobius:CYTOPLASMIC_DOMAIN,Pfam:PF07714,PIRSF:PIRSF000619,PROSITE_profiles:PS50011,PANTHER:PTHR24416,SMART:SM00219,Superfamily:SSF56112,CDD:cd05108",
                  "-",
                  "ENST00000275493.7:c.2573T>G",
                  "ENSP00000275493.2:p.Leu858Arg",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "drug_response,pathogenic",
                  "0,1,1",
                  "1,1,1",
                  "24944790,31019283,22992668,27618021,32295184,25157968,26619011,29525983,15118073,15118125,15329413,16115929,16204070,17192902,17285735,19922469,20479403,22753918,23102728,15897572,15901872,16043828,16203769,16204011,16865253,16956694,17047654,17106442,17317677,17387341,17429313,17473659,18303429,18349398,19096302,19455431,19692680,19692684,20022809,20038723,20573926,21670455,21783417,21900837,21969500,22215752,22370314,22452895,22740981,22760226,22982650,23816960,23948351,26490356,26515464,19147750,17877814,32175330,28347348,26053404,36388934,32962681,30149365,29721857,36629526,38008767,39273318,21132006,27032107,18408761,24439929,22285168,23982599,24868098,24736073,27102076",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-"
                ],
                [
                  "1",
                  "NC_000007.14_140753336_A/T",
                  "NC_000007.14:140753336",
                  "T",
                  "ENSG00000157764",
                  "ENST00000646891",
                  "Transcript",
                  "missense_variant",
                  "2025",
                  "1799",
                  "600",
                  "V/E",
                  "gTg/gAg",
                  "rs113488022,CM112509,COSV56056643,COSV56065204,COSV56080151",
                  "MODERATE",
                  "-",
                  "-1",
                  "-",
                  "SNV",
                  "BRAF",
                  "HGNC",
                  "HGNC:1097",
                  "protein_coding",
                  "YES",
                  "MANE_Select",
                  "NM_004333.6",
                  "-",
                  "-",
                  "P4",
                  "CCDS5863.1",
                  "ENSP00000493543",
                  "P15056.264",
                  "-",
                  "UPI000013DF26",
                  "-",
                  "1",
                  "deleterious_low_confidence(0)",
                  "probably_damaging(0.935)",
                  "15/18",
                  "-",
                  "PDB-ENSP_mappings:1uwh.A,PDB-ENSP_mappings:1uwh.B,PDB-ENSP_mappings:1uwj.A,PDB-ENSP_mappings:1uwj.B,PDB-ENSP_mappings:2fb8.A,PDB-ENSP_mappings:2fb8.B,PDB-ENSP_mappings:3c4c.A,PDB-ENSP_mappings:3c4c.B,PDB-ENSP_mappings:3d4q.A,PDB-ENSP_mappings:3d4q.B,PDB-ENSP_mappings:3idp.A,PDB-ENSP_mappings:3idp.B,PDB-ENSP_mappings:3ii5.A,PDB-ENSP_mappings:3ii5.B,PDB-ENSP_mappings:3og7.A,PDB-ENSP_mappings:3og7.B,PDB-ENSP_mappings:3ppj.A,PDB-ENSP_mappings:3ppj.B,PDB-ENSP_mappings:3ppk.A,PDB-ENSP_mappings:3ppk.B,PDB-ENSP_mappings:3prf.A,PDB-ENSP_mappings:3prf.B,PDB-ENSP_mappings:3pri.A,PDB-ENSP_mappings:3pri.B,PDB-ENSP_mappings:3psb.A,PDB-ENSP_mappings:3psb.B,PDB-ENSP_mappings:3psd.A,PDB-ENSP_mappings:3psd.B,PDB-ENSP_mappings:3q4c.A,PDB-ENSP_mappings:3q4c.B,PDB-ENSP_mappings:3q96.A,PDB-ENSP_mappings:3q96.B,PDB-ENSP_mappings:3skc.A,PDB-ENSP_mappings:3skc.B,PDB-ENSP_mappings:3tv4.A,PDB-ENSP_mappings:3tv4.B,PDB-ENSP_mappings:3tv6.A,PDB-ENSP_mappings:3tv6.B,PDB-ENSP_mappings:4cqe.A,PDB-ENSP_mappings:4cqe.B,PDB-ENSP_mappings:4dbn.A,PDB-ENSP_mappings:4dbn.B,PDB-ENSP_mappings:4e26.A,PDB-ENSP_mappings:4e26.B,PDB-ENSP_mappings:4e4x.A,PDB-ENSP_mappings:4e4x.B,PDB-ENSP_mappings:4ehe.A,PDB-ENSP_mappings:4ehe.B,PDB-ENSP_mappings:4ehg.A,PDB-ENSP_mappings:4ehg.B,PDB-ENSP_mappings:4fc0.A,PDB-ENSP_mappings:4fc0.B,PDB-ENSP_mappings:4fk3.A,PDB-ENSP_mappings:4fk3.B,PDB-ENSP_mappings:4g9c.A,PDB-ENSP_mappings:4g9c.B,PDB-ENSP_mappings:4g9r.A,PDB-ENSP_mappings:4g9r.B,PDB-ENSP_mappings:4h58.A,PDB-ENSP_mappings:4h58.B,PDB-ENSP_mappings:4h58.C,PDB-ENSP_mappings:4jvg.A,PDB-ENSP_mappings:4jvg.B,PDB-ENSP_mappings:4jvg.C,PDB-ENSP_mappings:4jvg.D,PDB-ENSP_mappings:4ksp.A,PDB-ENSP_mappings:4ksp.B,PDB-ENSP_mappings:4ksq.A,PDB-ENSP_mappings:4ksq.B,PDB-ENSP_mappings:4mbj.A,PDB-ENSP_mappings:4mbj.B,PDB-ENSP_mappings:4mne.B,PDB-ENSP_mappings:4mne.C,PDB-ENSP_mappings:4mne.F,PDB-ENSP_mappings:4mne.G,PDB-ENSP_mappings:4mnf.A,PDB-ENSP_mappings:4mnf.B,PDB-ENSP_mappings:4pp7.A,PDB-ENSP_mappings:4pp7.B,PDB-ENSP_mappings:4r5y.A,PDB-ENSP_mappings:4r5y.B,PDB-ENSP_mappings:4rzv.A,PDB-ENSP_mappings:4rzv.B,PDB-ENSP_mappings:4rzw.A,PDB-ENSP_mappings:4rzw.B,PDB-ENSP_mappings:4wo5.A,PDB-ENSP_mappings:4wo5.B,PDB-ENSP_mappings:4xv1.A,PDB-ENSP_mappings:4xv1.B,PDB-ENSP_mappings:4xv2.A,PDB-ENSP_mappings:4xv2.B,PDB-ENSP_mappings:4xv3.A,PDB-ENSP_mappings:4xv3.B,PDB-ENSP_mappings:4xv9.A,PDB-ENSP_mappings:4yht.A,PDB-ENSP_mappings:4yht.B,PDB-ENSP_mappings:5c9c.A,PDB-ENSP_mappings:5c9c.B,PDB-ENSP_mappings:5csw.A,PDB-ENSP_mappings:5csw.B,PDB-ENSP_mappings:5csx.A,PDB-ENSP_mappings:5ct7.A,PDB-ENSP_mappings:5ct7.B,PDB-ENSP_mappings:5fd2.A,PDB-ENSP_mappings:5fd2.B,PDB-ENSP_mappings:5hi2.A,PDB-ENSP_mappings:5hid.A,PDB-ENSP_mappings:5hid.B,PDB-ENSP_mappings:5hie.A,PDB-ENSP_mappings:5hie.B,PDB-ENSP_mappings:5hie.C,PDB-ENSP_mappings:5hie.D,PDB-ENSP_mappings:5ita.A,PDB-ENSP_mappings:5ita.B,PDB-ENSP_mappings:5jrq.A,PDB-ENSP_mappings:5jrq.B,PDB-ENSP_mappings:5jsm.A,PDB-ENSP_mappings:5jsm.B,PDB-ENSP_mappings:5jsm.C,PDB-ENSP_mappings:5jsm.D,PDB-ENSP_mappings:5jt2.A,PDB-ENSP_mappings:5jt2.B,PDB-ENSP_mappings:5jt2.C,PDB-ENSP_mappings:5jt2.D,PDB-ENSP_mappings:5val.A,PDB-ENSP_mappings:5val.B,PDB-ENSP_mappings:5vam.A,PDB-ENSP_mappings:5vam.B,PDB-ENSP_mappings:6b8u.A,PDB-ENSP_mappings:6b8u.B,PDB-ENSP_mappings:6cad.A,PDB-ENSP_mappings:6cad.B,PDB-ENSP_mappings:6n0p.A,PDB-ENSP_mappings:6n0p.B,PDB-ENSP_mappings:6n0q.A,PDB-ENSP_mappings:6n0q.B,PDB-ENSP_mappings:6nsq.A,PDB-ENSP_mappings:6nsq.B,PDB-ENSP_mappings:6nyb.A,PDB-ENSP_mappings:6p3d.A,PDB-ENSP_mappings:6p7g.A,PDB-ENSP_mappings:6p7g.B,PDB-ENSP_mappings:6p7g.C,PDB-ENSP_mappings:6p7g.D,PDB-ENSP_mappings:6pp9.A,PDB-ENSP_mappings:6q0j.A,PDB-ENSP_mappings:6q0j.B,PDB-ENSP_mappings:6q0k.A,PDB-ENSP_mappings:6q0k.B,PDB-ENSP_mappings:6q0t.A,PDB-ENSP_mappings:6q0t.B,PDB-ENSP_mappings:6u2g.B,PDB-ENSP_mappings:6u2h.C,PDB-ENSP_mappings:6u2h.D,PDB-ENSP_mappings:6uan.B,PDB-ENSP_mappings:6uan.C,PDB-ENSP_mappings:6uuo.A,PDB-ENSP_mappings:6uuo.B,PDB-ENSP_mappings:6v2u.A,PDB-ENSP_mappings:6v2u.B,PDB-ENSP_mappings:6v2w.A,PDB-ENSP_mappings:6v34.A,PDB-ENSP_mappings:6v34.B,PDB-ENSP_mappings:6xag.C,PDB-ENSP_mappings:6xag.D,PDB-ENSP_mappings:6xfp.A,PDB-ENSP_mappings:6xlo.A,PDB-ENSP_mappings:6xlo.B,PDB-ENSP_mappings:7k0v.A,PDB-ENSP_mappings:7k0v.B,PDB-ENSP_mappings:7k0v.C,PDB-ENSP_mappings:7k0v.D,PDB-ENSP_mappings:7m0t.A,PDB-ENSP_mappings:7m0u.A,PDB-ENSP_mappings:7m0v.A,PDB-ENSP_mappings:7m0w.A,PDB-ENSP_mappings:7m0x.A,PDB-ENSP_mappings:7m0y.A,PDB-ENSP_mappings:7m0z.A,PDB-ENSP_mappings:7mfd.A,PDB-ENSP_mappings:7mfe.A,PDB-ENSP_mappings:7mff.A,PDB-ENSP_mappings:7mff.B,PDB-ENSP_mappings:7p3v.A,PDB-ENSP_mappings:7p3v.B,PDB-ENSP_mappings:7shv.A,PDB-ENSP_mappings:7shv.B,PDB-ENSP_mappings:7zr0.K,PDB-ENSP_mappings:7zr5.K,PDB-ENSP_mappings:7zr6.K,PDB-ENSP_mappings:8c7x.A,PDB-ENSP_mappings:8c7x.B,PDB-ENSP_mappings:8c7y.A,PDB-ENSP_mappings:8c7y.B,PDB-ENSP_mappings:8dgs.A,PDB-ENSP_mappings:8dgt.A,PDB-ENSP_mappings:8f7o.A,PDB-ENSP_mappings:8f7o.B,PDB-ENSP_mappings:8f7p.A,PDB-ENSP_mappings:8f7p.B,PDB-ENSP_mappings:8qqg.A,PDB-ENSP_mappings:8qqg.B,PDB-ENSP_mappings:8qqg.C,PDB-ENSP_mappings:9axx.B,PDB-ENSP_mappings:9axx.D,PDB-ENSP_mappings:9axy.A,PDB-ENSP_mappings:9bfb.A,Superfamily:SSF56112,Pfam:PF07714,Gene3D:1.10.510.10,SMART:SM00220,PANTHER:PTHR44329,PROSITE_profiles:PS50011,CDD:cd14062,AFDB-ENSP_mappings:AF-P15056-F1",
                  "-",
                  "ENST00000646891.2:c.1799T>A",
                  "ENSP00000493543.1:p.Val600Glu",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "1.369e-06",
                  "0",
                  "0",
                  "0",
                  "0",
                  "0",
                  "0",
                  "9e-07",
                  "0",
                  "1.159e-05",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-",
                  "1.159e-05",
                  "gnomADe_SAS",
                  "not_provided,likely_pathogenic,pathogenic",
                  "0,0,1,1,1",
                  "1,1,1,1,1",
                  "23757202,30089490,26900293,12460918,20619739,21107323,22773810,23614898,25032700,25157968,26619011,24144365,31752122,12960123,12068308,12460919,14679157,15035987,19001320,19010912,19018267,19238210,19404918,19537845,19561230,20350999,20413299,20630094,20735442,20818844,21129611,21156289,21163703,21426297,21483012,21639808,21683865,21975775,22038996,22048237,22180495,22281684,22351686,22356324,22389471,22448344,22536370,22586120,22608338,22649091,22663011,22735384,22743296,22805292,22972589,22997239,23020132,23031422,23251002,23325582,23470635,23524406,23549875,23812671,23833300,23845441,23918947,24107445,24163374,24388723,24576830,24583796,24586605,24594804,25024077,25370471,25989278,26678033,12198537,21107320,33076847,32184228,26090869,32843432,32238401,26310374,28607096,35311178,31371350,30770838,26513174,28556791,32933095,24512911,32770181,36733350,28708103,28282860,35904569,35820816,30797065,26909593,34667063,36866106,36686473,29043205,35582142,24670642,15001635,15356022,23302800,15342696,12670889,1244737",
                  "-",
                  "-",
                  "-",
                  "-",
                  "-"
                ]
              ],
              "shape": {
                "columns": 82,
                "rows": 2
              }
            },
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Uploaded_variation</th>\n",
              "      <th>Location</th>\n",
              "      <th>Allele</th>\n",
              "      <th>Gene</th>\n",
              "      <th>Feature</th>\n",
              "      <th>Feature_type</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>cDNA_position</th>\n",
              "      <th>CDS_position</th>\n",
              "      <th>Protein_position</th>\n",
              "      <th>...</th>\n",
              "      <th>MAX_AF_POPS</th>\n",
              "      <th>CLIN_SIG</th>\n",
              "      <th>SOMATIC</th>\n",
              "      <th>PHENO</th>\n",
              "      <th>PUBMED</th>\n",
              "      <th>MOTIF_NAME</th>\n",
              "      <th>MOTIF_POS</th>\n",
              "      <th>HIGH_INF_POS</th>\n",
              "      <th>MOTIF_SCORE_CHANGE</th>\n",
              "      <th>TRANSCRIPTION_FACTORS</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>NC_000007.14_55191822_T/G</td>\n",
              "      <td>NC_000007.14:55191822</td>\n",
              "      <td>G</td>\n",
              "      <td>ENSG00000146648</td>\n",
              "      <td>ENST00000275493</td>\n",
              "      <td>Transcript</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>2834</td>\n",
              "      <td>2573</td>\n",
              "      <td>858</td>\n",
              "      <td>...</td>\n",
              "      <td>-</td>\n",
              "      <td>drug_response,pathogenic</td>\n",
              "      <td>0,1,1</td>\n",
              "      <td>1,1,1</td>\n",
              "      <td>24944790,31019283,22992668,27618021,32295184,2...</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>NC_000007.14_140753336_A/T</td>\n",
              "      <td>NC_000007.14:140753336</td>\n",
              "      <td>T</td>\n",
              "      <td>ENSG00000157764</td>\n",
              "      <td>ENST00000646891</td>\n",
              "      <td>Transcript</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>2025</td>\n",
              "      <td>1799</td>\n",
              "      <td>600</td>\n",
              "      <td>...</td>\n",
              "      <td>gnomADe_SAS</td>\n",
              "      <td>not_provided,likely_pathogenic,pathogenic</td>\n",
              "      <td>0,0,1,1,1</td>\n",
              "      <td>1,1,1,1,1</td>\n",
              "      <td>23757202,30089490,26900293,12460918,20619739,2...</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>2 rows Ã 82 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "           Uploaded_variation                Location Allele             Gene  \\\n",
              "0   NC_000007.14_55191822_T/G   NC_000007.14:55191822      G  ENSG00000146648   \n",
              "1  NC_000007.14_140753336_A/T  NC_000007.14:140753336      T  ENSG00000157764   \n",
              "\n",
              "           Feature Feature_type       Consequence  cDNA_position  \\\n",
              "0  ENST00000275493   Transcript  missense_variant           2834   \n",
              "1  ENST00000646891   Transcript  missense_variant           2025   \n",
              "\n",
              "   CDS_position  Protein_position  ...  MAX_AF_POPS  \\\n",
              "0          2573               858  ...            -   \n",
              "1          1799               600  ...  gnomADe_SAS   \n",
              "\n",
              "                                    CLIN_SIG    SOMATIC      PHENO  \\\n",
              "0                   drug_response,pathogenic      0,1,1      1,1,1   \n",
              "1  not_provided,likely_pathogenic,pathogenic  0,0,1,1,1  1,1,1,1,1   \n",
              "\n",
              "                                              PUBMED  MOTIF_NAME MOTIF_POS  \\\n",
              "0  24944790,31019283,22992668,27618021,32295184,2...           -         -   \n",
              "1  23757202,30089490,26900293,12460918,20619739,2...           -         -   \n",
              "\n",
              "  HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
              "0            -                  -                     -  \n",
              "1            -                  -                     -  \n",
              "\n",
              "[2 rows x 82 columns]"
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "import pandas as pd\n",
        "import numpy as np\n",
        "from io import StringIO\n",
        "\n",
        "# Robustly load VEP TSV so the header (the line starting with a single '#') is preserved\n",
        "vep_path = 'vep/caso3-tumor-pulmonar-host-variants.vep.tsv'\n",
        "with open(vep_path, 'r', encoding='utf-8') as fh:\n",
        "    lines = fh.readlines()\n",
        "\n",
        "# Find the header line that starts with a single '#'<not with '##'>\n",
        "header_idx = None\n",
        "for i, line in enumerate(lines):\n",
        "    if line.startswith('#') and not line.startswith('##'):\n",
        "        header_idx = i\n",
        "        break\n",
        "\n",
        "if header_idx is None:\n",
        "    # Fallback: try simple read (may fail if header was commented)\n",
        "    vep_df = pd.read_csv(vep_path, sep='\\t', comment='#')\n",
        "else:\n",
        "    header_line = lines[header_idx].lstrip('#').strip()\n",
        "    headers = header_line.split('\\t')\n",
        "    data = ''.join(lines[header_idx+1:])\n",
        "    vep_df = pd.read_csv(StringIO(data), sep='\\t', names=headers, header=None)\n",
        "\n",
        "# Summary\n",
        "print(f\"Total de variantes anotadas: {len(vep_df)}\")\n",
        "print('\\nColunas (primeiras 40):')\n",
        "print(list(vep_df.columns)[:40])\n",
        "\n",
        "# Show first rows\n",
        "vep_df.head(3)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Variantes de alto impacto: 2\n",
            "\n",
            "Genes afetados:\n",
            "SYMBOL\n",
            "EGFR    1\n",
            "BRAF    1\n",
            "Name: count, dtype: int64\n",
            "\n",
            "Detalhes das variantes de alto impacto:\n"
          ]
        },
        {
          "data": {
            "application/vnd.microsoft.datawrangler.viewer.v0+json": {
              "columns": [
                {
                  "name": "index",
                  "rawType": "int64",
                  "type": "integer"
                },
                {
                  "name": "Location",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "SYMBOL",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "Consequence",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "HGVSc",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "HGVSp",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "IMPACT",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "CANONICAL",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "AF",
                  "rawType": "object",
                  "type": "string"
                },
                {
                  "name": "MAX_AF",
                  "rawType": "object",
                  "type": "string"
                }
              ],
              "ref": "935c9d97-6fec-4212-be96-d7184d768cac",
              "rows": [
                [
                  "0",
                  "NC_000007.14:55191822",
                  "EGFR",
                  "missense_variant",
                  "ENST00000275493.7:c.2573T>G",
                  "ENSP00000275493.2:p.Leu858Arg",
                  "MODERATE",
                  "YES",
                  "-",
                  "-"
                ],
                [
                  "1",
                  "NC_000007.14:140753336",
                  "BRAF",
                  "missense_variant",
                  "ENST00000646891.2:c.1799T>A",
                  "ENSP00000493543.1:p.Val600Glu",
                  "MODERATE",
                  "YES",
                  "-",
                  "1.159e-05"
                ]
              ],
              "shape": {
                "columns": 9,
                "rows": 2
              }
            },
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Location</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>HGVSc</th>\n",
              "      <th>HGVSp</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>CANONICAL</th>\n",
              "      <th>AF</th>\n",
              "      <th>MAX_AF</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>NC_000007.14:55191822</td>\n",
              "      <td>EGFR</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000275493.7:c.2573T&gt;G</td>\n",
              "      <td>ENSP00000275493.2:p.Leu858Arg</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>YES</td>\n",
              "      <td>-</td>\n",
              "      <td>-</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>NC_000007.14:140753336</td>\n",
              "      <td>BRAF</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>ENST00000646891.2:c.1799T&gt;A</td>\n",
              "      <td>ENSP00000493543.1:p.Val600Glu</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>YES</td>\n",
              "      <td>-</td>\n",
              "      <td>1.159e-05</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                 Location SYMBOL       Consequence  \\\n",
              "0   NC_000007.14:55191822   EGFR  missense_variant   \n",
              "1  NC_000007.14:140753336   BRAF  missense_variant   \n",
              "\n",
              "                         HGVSc                          HGVSp    IMPACT  \\\n",
              "0  ENST00000275493.7:c.2573T>G  ENSP00000275493.2:p.Leu858Arg  MODERATE   \n",
              "1  ENST00000646891.2:c.1799T>A  ENSP00000493543.1:p.Val600Glu  MODERATE   \n",
              "\n",
              "  CANONICAL AF     MAX_AF  \n",
              "0       YES  -          -  \n",
              "1       YES  -  1.159e-05  "
            ]
          },
          "execution_count": 12,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "# Filtrar variantes de alto impacto (missense, frameshift, nonsense, etc.)\n",
        "# Focar em variantes nÃ£o-sinÃ´nimas que alteram a proteÃ­na\n",
        "pattern = r'missense|frameshift|stop_gained|stop_lost|start_lost|inframe'\n",
        "high_impact = vep_df[vep_df['Consequence'].str.contains(\n",
        "    pattern,\n",
        "    case=False,\n",
        "    na=False,\n",
        "    regex=True\n",
        ")]\n",
        "\n",
        "print(f\"\\nVariantes de alto impacto: {len(high_impact)}\")\n",
        "print(\"\\nGenes afetados:\")\n",
        "if 'SYMBOL' in high_impact.columns:\n",
        "    print(high_impact['SYMBOL'].value_counts())\n",
        "    \n",
        "# Exibir detalhes das variantes de alto impacto\n",
        "print(\"\\nDetalhes das variantes de alto impacto:\")\n",
        "columns_to_show = ['Location', 'SYMBOL', 'Consequence', 'HGVSc', 'HGVSp', \n",
        "                   'IMPACT', 'CANONICAL', 'AF', 'MAX_AF']\n",
        "available_cols = [col for col in columns_to_show if col in high_impact.columns]\n",
        "high_impact[available_cols]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Variantes encontradas em genes conhecidos de cÃ¢ncer de pulmÃ£o: 2\n",
            "\n",
            "Genes identificados:\n",
            "SYMBOL\n",
            "EGFR    1\n",
            "BRAF    1\n",
            "Name: count, dtype: int64\n",
            "\n",
            "================================================================================\n",
            "VARIANTES CLINICAMENTE RELEVANTES:\n",
            "================================================================================\n",
            "\n",
            "Gene: EGFR\n",
            "LocalizaÃ§Ã£o: NC_000007.14:55191822\n",
            "ConsequÃªncia: missense_variant\n",
            "HGVSc: ENST00000275493.7:c.2573T>G\n",
            "HGVSp: ENSP00000275493.2:p.Leu858Arg\n",
            "Impacto: MODERATE\n",
            "FrequÃªncia alÃ©lica (1000 Genomes): -\n",
            "--------------------------------------------------------------------------------\n",
            "\n",
            "Gene: BRAF\n",
            "LocalizaÃ§Ã£o: NC_000007.14:140753336\n",
            "ConsequÃªncia: missense_variant\n",
            "HGVSc: ENST00000646891.2:c.1799T>A\n",
            "HGVSp: ENSP00000493543.1:p.Val600Glu\n",
            "Impacto: MODERATE\n",
            "FrequÃªncia alÃ©lica (1000 Genomes): -\n",
            "--------------------------------------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "# Filtrar por genes conhecidos de cÃ¢ncer de pulmÃ£o\n",
        "cancer_genes = ['TP53', 'EGFR', 'KRAS', 'ALK', 'ROS1', 'BRAF', 'MET', \n",
        "                'RET', 'ERBB2', 'PIK3CA', 'STK11', 'KEAP1']\n",
        "\n",
        "if 'SYMBOL' in vep_df.columns:\n",
        "    cancer_variants = vep_df[vep_df['SYMBOL'].isin(cancer_genes)]\n",
        "    \n",
        "    if len(cancer_variants) > 0:\n",
        "        print(f\"Variantes encontradas em genes conhecidos de cÃ¢ncer de pulmÃ£o: {len(cancer_variants)}\")\n",
        "        print(\"\\nGenes identificados:\")\n",
        "        print(cancer_variants['SYMBOL'].value_counts())\n",
        "        \n",
        "        print(\"\\n\" + \"=\"*80)\n",
        "        print(\"VARIANTES CLINICAMENTE RELEVANTES:\")\n",
        "        print(\"=\"*80)\n",
        "        for idx, row in cancer_variants.iterrows():\n",
        "            print(f\"\\nGene: {row.get('SYMBOL', 'N/A')}\")\n",
        "            print(f\"LocalizaÃ§Ã£o: {row.get('Location', 'N/A')}\")\n",
        "            print(f\"ConsequÃªncia: {row.get('Consequence', 'N/A')}\")\n",
        "            print(f\"HGVSc: {row.get('HGVSc', 'N/A')}\")\n",
        "            print(f\"HGVSp: {row.get('HGVSp', 'N/A')}\")\n",
        "            print(f\"Impacto: {row.get('IMPACT', 'N/A')}\")\n",
        "            print(f\"FrequÃªncia alÃ©lica (1000 Genomes): {row.get('AF', 'N/A')}\")\n",
        "            print(\"-\"*80)\n",
        "    else:\n",
        "        print(\"Nenhuma variante encontrada nos genes de cÃ¢ncer de pulmÃ£o conhecidos.\")\n",
        "        print(\"Verificando todas as variantes...\")\n",
        "else:\n",
        "    print(\"Coluna SYMBOL nÃ£o encontrada. Exibindo todas as variantes:\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Interpretacao dos Resultados\n",
        "\n",
        "\n",
        "### Gene Identificado:\n",
        "- **EGFR**\n",
        "- **BRAF**\n",
        "\n",
        "### Variante Identificada:\n",
        "- **EGFR**\n",
        "    - **Nomenclatura HGVSc:** ENST00000275493.7:c.2573T>G\n",
        "    - **Nomenclatura HGVSp:** ENSP00000275493.2:p.Leu858Arg\n",
        "    - **Tipo de variante:** missense\n",
        "    - **LocalizaÃ§Ã£o:** NC_000007.14:55191822\n",
        "    - **Impacto:** MODERATE\n",
        "\n",
        "- **BRAF**\n",
        "    - **Nomenclatura HGVSc:** ENST00000646891.2:c.1799T>A\n",
        "    - **Nomenclatura HGVSp:** ENSP00000493543.1:p.Val600Glu\n",
        "    - **Tipo de variante:** missense\n",
        "    - **LocalizaÃ§Ã£o:** NC_000007.14:140753336\n",
        "    - **Impacto:** MODERATE\n",
        "\n",
        "### SignificÃ¢ncia ClÃ­nica:\n",
        "- **EGFR**\n",
        "    Variacao eh oncogenica, ocorrendo no Exon 21 do dominio da tirosina quinase. Deixa a TK constantemente ativada, independendo de vias de sinalizacao celular. Comum em casos de canceres de pulmao.\n",
        "    [Referencia](https://www.oncokb.org/gene/EGFR/L858R?refGenome=GRCh38)\n",
        "\n",
        "- **BRAF**\n",
        "    - Variacao eh oncogenica, ocorrendo no Exon 15 do dominio kinase, comum em casos de melanoma, e canceres de pulmao ou tireoide.\n",
        "    - Mutacao deixa a via BRAF constantemente ativada na forma monomerica, mas ainda sensiveis a inibidores RAF.\n",
        "    [Referencia](https://www.oncokb.org/gene/BRAF/V600E)\n",
        "\n",
        "\n",
        "\n",
        "### Conduta ClÃ­nica Sugerida:\n",
        "- **EGFR**\n",
        "    - Tratamento com inibidores de tirosina quinase de 1a,2a, ou 3a geracao\n",
        "    \n",
        "- **BRAF**\n",
        "    - Tratamento com inibidores de tirosina quinase, ou inibidores RAF\n",
        "\n",
        "### OpÃ§Ãµes TerapÃªuticas:\n",
        "- Tratamento de dupla acao, inibindo as TKs, junto do dominio MAPK do paciente\n",
        "    - Dabrafenib + Trametinib\n"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "env",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.14"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
